src/components/pages/Onboarding/Onboarding.tsx

```diff
@@ -0,0 +1,184 @@
+import * as React from "react";
+import { Member } from "../../../store/reducers/members";
+import { Text, View, StyleSheet } from "react-native";
+import { connect, MapStateToProps } from "react-redux";
+import { RahaState } from "../../../store";
+import { OnboardingSplash } from "./OnboardingSplash";
+import { VerifyName } from "./VerifyName";
+import { ChooseInviter } from "./ChooseInviter";
+import { OnboardingCamera } from "./OnboardingCamera";
+import { VideoPreview } from "./VideoPreview";
+
+/**
+ * Parent component for Onboarding flow.
+ */
+
+enum OnboardingStep {
+  SPLASH,
+  CHOOSE_INVITER,
+  VERIFY_NAME,
+  CAMERA,
+  VIDEO_PREVIEW,
+  REQUEST_INVITE
+}
+
+type ReduxStateProps = {
+  displayName: string | null;
+};
+
+type OwnProps = {
+  deeplinkInvitingMember?: Member;
+  navigation: any;
+};
+
+type OnboardingProps = ReduxStateProps & OwnProps;
+
+type OnboardingState = {
+  step: OnboardingStep;
+  invitingMember?: Member;
+  verifiedName?: string;
+  videoUri?: string;
+  errorMessage?: string;
+};
+
+export class OnboardingView extends React.Component<
```

Comment:
It just occurred to me, a shortcoming of this approach is that native back button on android for example wouldn’t work... if we care about that working then maybe the approach would be to do the page switching with a nested StackNavigator or something from React-navigation instead of via a switch statement. Still would be able to do this prop passing like it is here, but that way the nav native bindings should work?

Reply:
Will just close this PR out and keep the existing SignedOut StackNavigator.